api/src/routes/auth.ts

```diff
@@ -1,12 +1,112 @@
 import { Hono } from "hono";
-import { nanoid } from "nanoid";
+import { loginSchema, registerSchema } from "../validators/users.ts";
+import { db } from "../db/drizzle.ts";
+import { usersTable } from "../db/schema.ts";
+import {eq, or } from "drizzle-orm";
+import bcrypt from "bcryptjs";
+import { Jwt } from "hono/utils/jwt";
+import 'dotenv/config'
 
 const app = new Hono().basePath("/auth");
 
 app.post("/token", async (c) => {
-  const body = await c.req.json();
+  try{
+    const reqBody = await c.req.json()
 
-  return c.json({ data: { token: nanoid(), ...body } });
+    //validating the user input data
+    const result = loginSchema.safeParse(reqBody)
+    if(!result.success){
+      return c.json({
+        ok:false,
+        message:"invalid input data"
+      },404)
+    }
+
+    //geting the validated data
+    const {username, password} = result.data
+
+    //checking if the user exist
+    const user = await db.select().from(usersTable).where(eq(usersTable.username, username))
+    if(user.length === 0 ){
+      return c.json({
+        ok:false,
+        message:"invalid username",
+      }, 401)
+    }
+
+    //checking the password
+    const passwordMatch = await bcrypt.compare(password, user[0].password as string)
+    if(!passwordMatch){
+      return c.json({
+        ok:false,
+        message:"invalid password",
+      }, 401)
+    }
```

Comment:
all this check could be done in one `if` statement.
```js
if (!user || !(await argon2.verify(String(user.password), output.password))) {
      return c.json({ success: false, error: "invalid username or password" }, 400);
    }
```
> the above codebase was using argon2 instead of bcrypt.